dbt/adapters/snowflake/relation_configs/target_lag.py

```diff
@@ -26,3 +39,101 @@ class SnowflakeDynamicTableTargetLagConfig(RelationConfigBase):
 
     duration: int
     period: SnowflakeDynamicTableTargetLagPeriod
+
+    @property
+    def validation_rules(self) -> Set[RelationConfigValidationRule]:
+        return {
+            RelationConfigValidationRule(
+                validation_check=self.duration is not None and self.period is not None,
+                validation_error=DbtRuntimeError(
+                    "Target lag for a materialized view requires both a duration and a period."
+                ),
+            ),
+            RelationConfigValidationRule(
+                validation_check=(
+                    (
+                        self.duration >= 1
+                        and self.period != SnowflakeDynamicTableTargetLagPeriod.seconds
+                    )
+                    or (
+                        self.duration >= 60
+                        and self.period == SnowflakeDynamicTableTargetLagPeriod.seconds
+                    )
+                ),
+                validation_error=DbtRuntimeError(
+                    f"The minimum for target lag for a materialized view is 1 minute. "
+                    f"The provided value is {self.duration} {self.period.value}."
+                ),
+            ),
+        }
+
+    @classmethod
+    def from_dict(cls, config_dict: dict) -> "SnowflakeDynamicTableTargetLagConfig":
+        kwargs_dict: Dict[str, Union[int, SnowflakeDynamicTableTargetLagPeriod]] = {}
+
+        if duration := config_dict.get("duration"):
+            kwargs_dict.update({"duration": int(duration)})
+
+        if period := config_dict.get("period"):
+            kwargs_dict.update({"period": SnowflakeDynamicTableTargetLagPeriod(period)})
+
+        target_lag: "SnowflakeDynamicTableTargetLagConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return target_lag
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
+        target_lag: str = model_node.config.extra.get("target_lag")
+        try:
+            duration, period = target_lag.split(" ")
+        except IndexError:
```

Comment:
Should also catch the `AttributeError` when `target_lag` is `None` (since `get` can do that) unless that's guarded against elsewhere.

Reply:
It might be in `dbt/adapters/snowflake/relation_configs/dynamic_table.py` around line `78`? Still, if you ever instantiate this without the `AttributeError` check, you'll get an exception.

Reply:
Yeah, that sounds right. I think I originally had this in an if:
```python
if target_lag := model_node.config.extra.get("target_lag"):
    try:
        duration, period = target_lag.split(" ")
    except IndexError:
        duration, period = None, None
```
But then I removed the if and forgot about that case. I'll update.